### DISCUSSION TITLE
CI/CD

### DISCUSSION THREAD

**Discussion by sanderdekroon** [OpenWebconcept/Afspraken#3] [General]
Hoe gaan de CI/CD straat voor Open Webconcept vormgeven? 

De meeste plugins hebben PHPUnit tests die we bij MR/PR's willen draaien. Op welk platform gaan we dit doen? 
- Github Actions
- Circle CI
- Ander platform?

In de toekomst willen we wellicht ook meer doen dan alleen (unit)tests draaien. Bijvoorbeeld:
- Vulnerability scanning
- Code kwaliteit controle
- Code formatting
- Releases / productie builds maken & uploaden

**Comment by MWest2020** [comment]
Github Actions is in elk geval goedkoper. Nadeel is dat de UI een stuk minder is. Een voordeel dat het iets minder set-up heeft en het kan changelog automatisch genereren. Voor simpele CI/ CD is GHA voldoende.

Als we in de toekomst meer met de CI/CD straat gaan doen, dan lijkt me dat Circle CI nodig is.
